packages/app/src/client/utils/request.ts

```diff
@@ -0,0 +1,16 @@
+import axios from "axios";
+import * as rax from 'retry-axios';
+
+const request = axios.create({
+  // For config options see https://github.com/axios/axios#request-config
+  timeout: 0,
```

Comment:
Let's set this to something non-zero, i.e. let's find a timeout constant we're comfortable with here. As this affects various requests, there is generally no one-size-fits-all timeout constant. But maybe (and you know best) there is.

If requests are small (just a few bytes) and if the response generation is known to be fast (sub-second) then let's set this to `5 s` or `10 s` because then we "know" that something is wrong, and retrying after 5 to 10 seconds already might heal the situation.

Reply:
Completely agree.

Reply:
One thing regarding that, that is not quite in the scope of this is, that we need to do a better job in communicating these possible large request times to the user.
It might not be obvious to the user that the request has still a chance of succeeding after 5s. We might want to choose to show a spinner (or any other kind of "loading" state) in the UI or in certain situations. In some situations we do not seem to do that.

Reply:
> We might want to choose to show a spinner (or any other kind of "loading" state) in the UI or in certain situations

100 % agree. For anything that's expected to take longer than N seconds, show a spinner or something like that.

Reply:
yes fully agree!

Reply:
> In some situations we do not seem to do that

Not because we don't want to, just haven't got there yet!